src/Education/XamlGenerator/RuleHelpXamlTranslator.cs

```diff
@@ -0,0 +1,459 @@
+        /// </summary>
+        private bool tableAlternateRow;
+
+        public string TranslateHtmlToXaml(string htmlContent)
+        {
+            var sb = new StringBuilder();
+            writer = CreateXmlWriter(sb);
+            reader = CreateXmlReader(htmlContent);
+            outputXamlElementStack = new Stack<XamlOutputElementInfo>();
+
+            //We are putting this to simulate the root which will accept only Blocks. So we can add paragraph to inline elements to make them compatible.
+            outputXamlElementStack.Push(new XamlOutputElementInfo("xaml root", false));
+
+            try
+            {
+                while (reader.Read())
+                {
+                    switch (reader.NodeType)
+                    {
+                        case XmlNodeType.Element:
+                            ProcessElement();
+                            break;
+
+                        case XmlNodeType.Text:
+                            WriteText(reader.Value);
+                            break;
+
+                        case XmlNodeType.Whitespace:
+                            writer.WriteString(reader.Value);
+                            break;
+
+                        case XmlNodeType.EndElement:
+                            XamlOutputElementInfo xamlOutputElement;
+                            do
+                            {
+                                xamlOutputElement = outputXamlElementStack.Pop();
+                                writer.WriteEndElement();
+                            } while (xamlOutputElement.HtmlElementName != reader.Name);
+
+                            break;
+
+                        default:
+                            var message = string.Format(Resources.XamlBuilder_UnexpectedNodeError, reader.NodeType, reader.Name, reader.Value);
+                            throw new InvalidDataException(message);
+                    }
+                }
+
+                // We've processed all of the html elements.
+                // Now, the stack should only contain the root FlowDocument element, plus an extra
+                // block element if the first tag we processed was an Inline.
+                Debug.Assert((outputXamlElementStack.Count <= 2) || (outputXamlElementStack.Count <= 1), "Expecting at most 2 unclosed elements in the stack");
+                Debug.Assert((outputXamlElementStack.Count == 1 && outputXamlElementStack.Peek().HtmlElementName == "xaml root")
+                    || (outputXamlElementStack.Count == 2 &&
+                        outputXamlElementStack.ToArray()[0].HtmlElementName == null &&
+                        outputXamlElementStack.ToArray()[1].HtmlElementName == "xaml root"),
+                        "Unexpected items in final stack");
+
+                //root element will be on the calling class so opening and closing it should be handled there
+                while (outputXamlElementStack.Count > 1)
```

Comment:
Change from the old version of RuleHelpXamlBuilder. we are not closing the root because we do know it. It should be handled by the caller.